src/model_tool.py

```diff
@@ -51,25 +50,48 @@ def compute_auc(y_true, y_pred):
 class ToxModel():
     def __init__(self, 
                  model_name = None, 
-                 model_dir = DEFAULT_MODEL_DIR):
+                 model_dir = DEFAULT_MODEL_DIR,
+                 hparams = None):
         self.model_dir = model_dir
         self.model_name = model_name
         self.model = None
         self.tokenizer = None
+        self.hparams = DEFAULT_HPARAMS
+        if hparams:
+            self.update_hparams(hparams)
         if model_name:
             self.load_model_from_name(model_name)
+        self.print_hparams()
+
+    def print_hparams(self):
+        print('Hyperparameters')
+        print('---------------')
+        for k, v in self.hparams.iteritems():
+            print('{}: {}'.format(k, v))
+        print('')
+
+    def update_hparams(self, new_hparams):
+        self.hparams.update(new_hparams)
         
     def get_model_name(self):
         return self.model_name
 
+    def save_hparams(self, model_name):
+        self.hparams['model_name'] = model_name
+        cPickle.dump(self.hparams, 
+            open(os.path.join(self.model_dir, 
+                '%s_hparams.pkl' % self.model_name), 'wb'))
```

Comment:
I think pickle is generally discouraged: http://www.benfrederickson.com/dont-pickle-your-data/

This data is pretty simple. Can we save it as JSON instead? Also makes it nicely human readable.

Reply:
Done.